nft-client/src/main.rs

```diff
@@ -0,0 +1,635 @@
+                &std::fs::read_to_string(sender).context("Could not read the keys file.")?,
+            )
+            .context("Could not parse the accounts file.")?;
+
+            let transfers = token_ids
+                .iter()
+                .map(|token_id| cts1::Transfer {
+                    token_id:     token_id.clone(),
+                    amount:       1,
+                    from:         convert_address(from.0.clone()),
+                    to:           convert_address(to.0.clone()),
+                    receive_name: to_func.as_ref().map(|wrapper| wrapper.clone().0),
+                    data:         to_func_data.clone(),
+                })
+                .collect();
+
+            let parameter = cts1::TransferParams(transfers);
+            let bytes = concordium_contracts_common::to_bytes(&parameter);
+
+            let payload = transactions::UpdateContractPayload {
+                amount:       common::types::Amount::from(0),
+                address:      contract.0,
+                receive_name: smart_contracts::ReceiveName::try_from(format!(
+                    "{}.transfer",
+                    NFT_CONTRACT_NAME
+                ))
+                .map_err(|e| anyhow!("Failed to parse receive name {}", e))?,
+                message:      smart_contracts::Parameter::from(bytes),
+            };
+
+            let transaction_payload = transactions::Payload::Update { payload };
+            println!(
+                "Transferring tokens {:?} from {} to {:?}",
+                token_ids, from, to
+            );
+            let hash =
+                send_transaction(&mut grpc_client, &account_data, transaction_payload).await?;
+            println!("Transaction send {}", hash);
+        }
+    }
+
+    Ok(())
+}
+
+/// Collect every contract event in a transaction summary for a given contract
+/// address.
+fn collect_summary_contract_events(
+    entry: DatabaseSummaryEntry,
+    contract: types::ContractAddress,
+) -> Vec<smart_contracts::ContractEvent> {
+    if let DatabaseSummaryEntry::BlockItem(item) = entry {
+        if let types::BlockItemSummaryDetails::AccountTransaction(tx) = item.details {
+            if let types::AccountTransactionEffects::ContractUpdateIssued { effects } = tx.effects {
+                let mut events = Vec::new();
+                for effect in effects {
+                    if let types::ContractTraceElement::Updated { data } = effect {
+                        if data.address == contract {
+                            events.extend(data.events)
+                        }
+                    }
+                }
+                return events;
+            } else if let types::AccountTransactionEffects::ContractInitialized { data } =
+                tx.effects
+            {
+                if data.address == contract {
+                    return data.events;
+                }
+            }
+        }
+    }
+    Vec::new()
```

Comment:
Recommend 
```rust
else {
    Vec::new()
}
```
and adding another else. The control flow is quite unclear in this function with so many nested conditionals.